### ISSUE TITLE
Android deeplink results in crash

### ISSUE THREAD

**Issue by vuurbeving** [MvvmCross/MvvmCross#3161]
## 🐛 Bug Report

If the app is running, deeplink works fine. If the app is not running and a deeplink is used to open the app, it will crash. After some investigation I think it all boils down to some code in `FindSetupType<TSetup>()` in MvxSetupExtensions. It tries to locate a type with name "Setup", which might not be available, since it is no longer required to use a seperate Setup class and in my case I don't have one.

Adding a Setup class prevents the crash, but doesn't follow the deeplink though.

Stacktrace (from logcat):

```
10-12 09:51:12.210      ZTE A2017U   Error       23476        AndroidRuntime   Parameter name: logger
  at MvvmCross.Logging.MvxLogExtensions.IsWarnEnabled (MvvmCross.Logging.IMvxLog logger) [0x0000d] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Logging.MvxLogExtensions.WarnFormat (MvvmCross.Logging.IMvxLog logger, System.String message, System.Object[] args) [0x00000] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Logging.MvxLogExtensions.Warn (MvvmCross.Logging.IMvxLog logger, System.String message, System.Object[] args) [0x00000] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.IoC.MvxTypeExtensions.ExceptionSafeGetTypes (System.Reflection.Assembly assembly) [0x0002b] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Core.MvxSetupExtensions+<>c__4`1[TSetup].<FindSetupType>b__4_0 (System.Reflection.Assembly assembly) [0x00000] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at System.Linq.Enumerable+<SelectManyIterator>d__167`3[TSource,TCollection,TResult].MoveNext () [0x0004e] in <3ff1b4644edb4d19b12e736b81734c89>:0 
  at System.Linq.Enumerable+WhereSelectEnumerableIterator`2[TSource,TResult].MoveNext () [0x00059] in <3ff1b4644edb4d19b12e736b81734c89>:0 
  at System.Linq.Enumerable.TryGetFirst[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Boolean& found) [0x00045] in <3ff1b4644edb4d19b12e736b81734c89>:0 
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00000] in <3ff1b4644edb4d19b12e736b81734c89>:0 
  at MvvmCross.Core.MvxSetupExtensions.FindSetupType[TSetup] () [0x000ba] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Core.MvxSetupExtensions.CreateSetup[TSetup] () [0x00000] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Core.MvxSetup.Instance () [0x00015] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Core.MvxSetupSingleton.CreateSetup () [0x00000] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
   --- End of inner exception stack trace ---
  at MvvmCross.Core.MvxSetupSingleton.CreateSetup () [0x00019] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Core.MvxSetupSingleton.EnsureSingletonAvailable[TMvxSetupSingleton] () [0x00046] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Platforms.Android.Core.MvxAndroidSetupSingleton.EnsureSingletonAvailable (Android.Content.Context applicationContext) [0x00000] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Platforms.Android.Views.MvxActivityViewExtensions.EnsureSetupInitialized (MvvmCross.Platforms.Android.Views.IMvxAndroidView androidView) [0x00016] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Platforms.Android.Views.MvxActivityViewExtensions.OnViewCreate (MvvmCross.Platforms.Android.Views.IMvxAndroidView androidView, Android.OS.Bundle bundle) [0x00014] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Platforms.Android.Views.MvxActivityAdapter.EventSourceOnCreateCalled (System.Object sender, MvvmCross.Base.MvxValueEventArgs`1[T] eventArgs) [0x0000c] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at (wrapper delegate-invoke) System.EventHandler`1[MvvmCross.Base.MvxValueEventArgs`1[Android.OS.Bundle]].invoke_void_object_TEventArgs(object,MvvmCross.Base.MvxValueEventArgs`1<Android.OS.Bundle>)
  at MvvmCross.Base.MvxDelegateExtensions.Raise[T] (System.EventHandler`1[TEventArgs] eventHandler, System.Object sender, T value) [0x0000b] in <8beeb96d9bc848278862c6e9b1d194c5>:0 
  at MvvmCross.Droid.Support.V7.AppCompat.EventSource.MvxEventSourceAppCompatActivity.OnCreate (Android.OS.Bundle bundle) [0x00014] in <ca04e443302e44bd8f23c508ae467fd1>:0 
  at MvvmCross.Droid.Support.V7.AppCompat.MvxAppCompatActivity.OnCreate (Android.OS.Bundle bundle) [0x00000] in <ca04e443302e44bd8f23c508ae467fd1>:0 
  at UniversalConsumerApp.Droid.Views.PointOfInterestDetailView.OnCreate (Android.OS.Bundle bundle) [0x00001] in <b0f27342bddc464b8b2446b4a48292bf>:0 
  at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x00011] in <9c89490198b4401e89a0f0bf9e68c7be>:0 
  at (wrapper dynamic-method) System.Object.3(intptr,intptr,intptr)
        at md51afdbe96323c214c47787078a85e8164.PointOfInterestDetailView.n_onCreate(Native Method)
        at md51afdbe96323c214c47787078a85e8164.PointOfInterestDetailView.onCreate(PointOfInterestDetailView.java:29)
        at android.app.Activity.performCreate(Activity.java:6684)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6186)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
  Force finishing activity UniversalConsumerApp.Droid.UniversalConsumerApp.Droid/md51afdbe96323c214c47787078a85e8164.PointOfInterestDetailView
  Force finishing activity org.mozilla.firefox/org.mozilla.gecko.BrowserApp
```

### Expected behavior

App launches and follows deeplink to the correct view.

### Reproduction steps

- Have a mvvmcross android app without Setup class.
ex. `public class SplashScreen : MvxSplashScreenAppCompatActivity<MvxAppCompatSetup<Core.App>, Core.App>` 
- Make sure the app is not running
- Open a registered deeplink
- Crash

### Configuration

**Version:** 6.2

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by Cheesebaron** [comment]
ExceptionSafeGetTypes blowing up, should be fixed in 6.2.1 see https://github.com/MvvmCross/MvvmCross/pull/3153

**Comment by vuurbeving** [comment]
Ok, I just tested this again with 6.2.1 and the exception is definitely different (so something got fixed i guess), but the exception thrown now is even more in line with the original problem (Setup class non-existant):

```
Time	Device Name	Type	PID	Tag	Message
10-12 15:52:14.761	ZTE A2017U	Error	5232	AndroidRuntime	android.runtime.JavaProxyThrowable: MvvmCross.Exceptions.MvxException: Failed to create setup instance ---> MvvmCross.Exceptions.MvxException: Could not find a Setup class for application
  at MvvmCross.Core.MvxSetupExtensions.CreateSetup[TSetup] () [0x00019] in <103d59106f92414bb3d9c03f45250bec>:0 
  at MvvmCross.Core.MvxSetup.Instance () [0x00015] in <103d59106f92414bb3d9c03f45250bec>:0 
  at MvvmCross.Core.MvxSetupSingleton.CreateSetup () [0x00000] in <103d59106f92414bb3d9c03f45250bec>:0 
   --- End of inner exception stack trace ---
  at MvvmCross.Core.MvxSetupSingleton.CreateSetup () [0x00017] in <103d59106f92414bb3d9c03f45250bec>:0 
  at MvvmCross.Core.MvxSetupSingleton.EnsureSingletonAvailable[TMvxSetupSingleton] () [0x00045] in <103d59106f92414bb3d9c03f45250bec>:0 
  at MvvmCross.Platforms.Android.Core.MvxAndroidSetupSingleton.EnsureSingletonAvailable (Android.Content.Context applicationContext) [0x00000] in <103d59106f92414bb3d9c03f45250bec>:0 
  at MvvmCross.Platforms.Android.Views.MvxActivityViewExtensions.EnsureSetupInitialized (MvvmCross.Platforms.Android.Views.IMvxAndroidView androidView) [0x00014] in <103d59106f92414bb3d9c03f45250bec>:0 
  at MvvmCross.Platforms.Android.Views.MvxActivityViewExtensions.OnViewCreate (MvvmCross.Platforms.Android.Views.IMvxAndroidView androidView, Android.OS.Bundle bundle) [0x00014] in <103d59106f92414bb3d9c03f45250bec>:0 
  at MvvmCross.Platforms.Android.Views.MvxActivityAdapter.EventSourceOnCreateCalled (System.Object sender, MvvmCross.Base.MvxValueEventArgs`1[T] eventArgs) [0x0000c] in <103d59106f92414bb3d9c03f45250bec>:0 
  at (wrapper delegate-invoke) System.EventHandler`1[MvvmCross.Base.MvxValueEventArgs`1[Android.OS.Bundle]].invoke_void_object_TEventArgs(object,MvvmCross.Base.MvxValueEventArgs`1<Android.OS.Bundle>)
  at MvvmCross.Base.MvxDelegateExtensions.Raise[T] (System.EventHandler`1[TEventArgs] eventHandler, System.Object sender, T value) [0x0000b] in <103d59106f92414bb3d9c03f45250bec>:0 
  at MvvmCross.Droid.Support.V7.AppCompat.EventSource.MvxEventSourceAppCompatActivity.OnCreate (Android.OS.Bundle bundle) [0x00014] in <6f4d53f489084371969126ea5e21accf>:0 
  at MvvmCross.Droid.Support.V7.AppCompat.MvxAppCompatActivity.OnCreate (Android.OS.Bundle bundle) [0x00000] in <6f4d53f489084371969126ea5e21accf>:0 
  at UniversalConsumerApp.Droid.Views.PointOfInterestDetailView.OnCreate (Android.OS.Bundle bundle) [0x00001] in <f10cdb88d3db4f1981176ee655185739>:0 
  at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x00011] in <9c89490198b4401e89a0f0bf9e68c7be>:0 
  at (wrapper dynamic-method) System.Object.3(intptr,intptr,intptr)
	at md51afdbe96323c214c47787078a85e8164.PointOfInterestDetailView.n_onCreate(Native Method)
	at md51afdbe96323c214c47787078a85e8164.PointOfInterestDetailView.onCreate(PointOfInterestDetailView.java:29)
	at android.app.Activity.performCreate(Activity.java:6684)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6186)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
  Force finishing activity UniversalConsumerApp.Droid.UniversalConsumerApp.Droid/md51afdbe96323c214c47787078a85e8164.PointOfInterestDetailView
```

I tried adding a Setup class again and that is a working workaround for the crash, i.e. the app starts, but the deeplink is not followed (it is followed properly when the app is already running before opening the deeplink)
